dirty_equals/_other.py

```diff
@@ -260,13 +262,15 @@ def equals(self, other: Any) -> bool:
             parsed = self.parse_obj_as(self.url_type, other)
         except self.ValidationError:
             raise ValueError('Invalid URL')
+
+        equal = parsed == other if self.pydantic_version < '2' else parsed.unicode_string() == other
```

Comment:
we'll have problems when pydantic hits V10!

Reply:
I fixed this problem here: https://github.com/samuelcolvin/dirty-equals/pull/80 😄